### PR TITLE
Added connection state IDLE

### PR DESCRIPTION
**Pull request by ktind**
Changed protobuf model to be more generic - Renamed G4Download to Download. Added receiver status to the download message

### CODE REVIEW

File: core/src/main/java/com/nightscout/core/drivers/AbstractDevice.java [hunk 1]
```diff
@@ -67,6 +70,14 @@ protected void onConnect() {
                 messages.getString("g4_connected"));
         connectionStatus = G4ConnectionState.CONNECTED;
         postConnectionStatus();
+        receiverState = new ReceiverState(new DateTime().getMillis(), ReceiverStatus.RECEIVER_CONNECTED);
+//        postReceiverState(receiverState);
```
Comment (trhodeos) [inline core/src/main/java/com/nightscout/core/drivers/AbstractDevice.java]: Uncomment this, or remove

File: core/src/main/java/com/nightscout/core/drivers/DexcomG4.java [hunk 2]
```diff
@@ -43,6 +43,8 @@
     protected String transmitterId = "";
     protected List<SensorRecord> lastSensorRecords;
     protected List<CalRecord> lastCalRecords;
+    private Bus bus;
+
```
Comment (trhodeos) [inline core/src/main/java/com/nightscout/core/drivers/DexcomG4.java:47]: Remove extra line.

File: core/src/main/java/com/nightscout/core/drivers/DexcomG4.java [hunk 3]
```diff
@@ -123,7 +126,7 @@ protected G4Download doDownload() {
 
         List<SensorGlucoseValueEntry> cookieMonsterG4SGVs;
         List<SensorEntry> cookieMonsterG4Sensors;
-        Bus bus = BusProvider.getInstance();
+//        Bus bus = BusProvider.getInstance();
```
Comment (trhodeos) [inline core/src/main/java/com/nightscout/core/drivers/DexcomG4.java]: Remove

**Comment by trhodeos** [comment]
Added a few comments. :+1:  otherwise. Also, travis is being weird. Not sure why one is passing, but the other is failing. Also, not sure why we have two.

**Comment by jasoncalabrese** [comment]
2 travis checks are running on crm too, would be nice to clear that up

**Pull request closed without merging**